src/Outsystems.SetupTools/Lib/PlatformSetup.ps1

```diff
@@ -79,28 +79,32 @@ function GetDotNet4Version()
 
     return $output
 }
-
-function GetDotNetCoreVersion()
+function GetWindowsServerHostingVersion()
 {
-    LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Stream 2 -Message "Getting the contents of the registry key HKLM:SOFTWARE\dotnet\Setup\InstalledVersions\x64\sharedhost\Version"
-
-    try
+    LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Stream 2 -Message "Getting the contents of the registry key HKLM:SOFTWARE\WOW6432Node\Microsoft\Updates\.NET Core\Microsoft .NET Core 2.1.11 - Windows Server Hosting (x64)\PackageVersion"
+    $DotNETCoreUpdatesPath = "HKLM:SOFTWARE\Wow6432Node\Microsoft\Updates\.NET Core"
+    $version = '0.0.0.0'
+    $PathExists = Test-Path $DotNETCoreUpdatesPath
+    if (-not $PathExists)
     {
-        $output = $(Get-ItemProperty -Path "HKLM:SOFTWARE\dotnet\Setup\InstalledVersions\x64\sharedhost" -Name "Version" -ErrorAction Stop).Version
+        LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Stream 2 -Message "Registry key " + $DotNETCoreUpdatesPath + " does not exist"
+        return $version
     }
-    catch
+    $DotNetCoreItems = Get-Item -ErrorAction Stop -Path $DotNETCoreUpdatesPath
```

Comment:
-ErrorAction Stop shouldn't be here. This internal functions should not throw. They should always return some value and the caller will chose what to do,
But I will check this better tomorrow morning.

Reply:
```suggestion
    $DotNetCoreItems = Get-Item -ErrorAction SilentlyContinue -Path $DotNETCoreUpdatesPath
```